filechanges/opt/musicbox/startup.sh

```diff
@@ -91,16 +91,31 @@ chmod u+s /sbin/shutdown
 
 if [ "$INI__network__wifi_network" != "" ]
 then
+
     #put wifi settings for wpa roaming
-cat >/etc/wpa.conf <<EOF
-    ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
-    update_config=1
-    network={
-        ssid="$INI__network__wifi_network"
-        psk="$INI__network__wifi_password"
-        scan_ssid=1
-    }
-EOF
+    if [ "$INI__network__wifi_password" != "" ]
+    then
+        cat >/etc/wpa.conf <<EOF
+            ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
+            update_config=1
+            network={
+                ssid="$INI__network__wifi_network"
+                psk="$INI__network__wifi_password"
+                scan_ssid=1
+            }
+        EOF
+    else
+        cat >/etc/wpa.conf <<EOF
+            ctrl_interface=DIR=/var/run/wpa_supplicant GROUP=netdev
+            update_config=1
+            network={
+                ssid="$INI__network__wifi_network"
+                proto=RSN
```

Comment:
Don't get me wrong, supporting open WiFi networks is something I've been meaning to get around to (hence #286). My comment regarding line 113 is specifically about the need for `proto=RSN`. Looking at the wpa_supplicant manual I wouldn't have thought it was necessary.